subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -293,3 +310,132 @@ export function handleGovSetReserveFee(event: GovSetReserveFee): void {
     event,
   );
 }
+
+export function handleStake(event: Stake): void {
+  const underlying = event.params.underlying.toHexString();
+  // find market id for underlying
+  const assetStatus = getOrCreateAssetStatus(underlying);
+  const marketId = assetStatus.eToken!;
+  const market = getOrCreateMarket(marketId);
+  const deltaStakedAmount = getDeltaStakeAmount(event);
+
+  // keep track of staked amount from epoch 1
+  const epochID = getCurrentEpoch(event);
+  if (epochID < 0) {
+    market._stakedAmount = market._stakedAmount.plus(deltaStakedAmount);
+    market.save();
+    return;
+  }
+
+  const epochStartBlock = getStartBlockForEpoch(epochID)!;
+  let epoch = _Epoch.load(epochID.toString());
+  if (!epoch) {
```

Comment:
This could be an early return statement as well:

```
if (epoch) {
    log;
    return;
}
```

Reply:
If I do this, I will need to repeat [lines 433-440](https://github.com/tnkrxyz/subgraphs/blob/16d7fb2e879f7c9cea658ebb2a8cb3996db1143f/subgraphs/euler-finance/src/mappings/handlers.ts#L433-L440)
```
  if (!market._stakeLastUpdateBlock) {
    market._stakeLastUpdateBlock = epochStartBlock;
    market._weightedStakedAmount = BIGINT_ZERO;
  }
  // update _weightedStakeAmount before updating _stakedAmount
  updateWeightedStakedAmount(market, event.block.number);
  market._stakedAmount = market._stakedAmount.plus(deltaStakedAmount);
  market.save();
```

And these lines have to be after the [new epoch logic (`if (!epoch) {`)](https://github.com/tnkrxyz/subgraphs/blob/16d7fb2e879f7c9cea658ebb2a8cb3996db1143f/subgraphs/euler-finance/src/mappings/handlers.ts#L332)

Reply:
Fair, I imagine it may change a little as well when we implement the new logic?

Reply:
Correct.